k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/ImapFolderPusher.java

```diff
@@ -253,6 +253,8 @@ private void reacquireWakeLockAndCleanUp() {
             } catch (Exception me) {
                 Log.e(LOG_TAG, "Got exception while closing for exception for " + getLogId(), me);
             }
+
+            connection = null;
```

Comment:
seems like this might lead to a null pointer exception on the very same close operation. which is inside a catch-all, but relying on that still feels ugly. is that intentional?